crates/db_views_actor/src/community_export_view.rs

```diff
@@ -0,0 +1,223 @@
+
+    let mut query = community::table
+      .left_join(local_user::table.on(local_user::person_id.eq(person_id_join)))
+      .left_join(
+        community_follower::table.on(
+          community::id
+            .eq(community_follower::community_id)
+            .and(community_follower::person_id.eq(person_id_join)),
+        ),
+      )
+      .left_join(
+        community_block::table.on(
+          community::id
+            .eq(community_block::community_id)
+            .and(community_block::person_id.eq(person_id_join)),
+        ),
+      )
+      .select((
+        community::all_columns,
+        community_follower::all_columns.nullable(),
+        community_block::all_columns.nullable(),
+      ))
+      .into_boxed();
+
+    if let Some(search_term) = self.search_term {
+      let searcher = fuzzy_search(&search_term);
+      query = query
+        .filter(community::name.ilike(searcher.clone()))
+        .or_filter(community::title.ilike(searcher));
+    };
+
+    // Hide deleted and removed for non-admins or mods
+    if !self.is_mod_or_admin.unwrap_or(false) {
+      query = query
+        .filter(community::removed.eq(false))
+        .filter(community::deleted.eq(false))
+        .filter(
+          community::hidden
+            .eq(false)
+            .or(community_follower::person_id.eq(person_id_join)),
+        );
+    }
+    match self.sort.unwrap_or(Old) {
+      New => query = query.order_by(community::published.desc()),
+      Old => query = query.order_by(community::published.asc()),
+    };
+
+    if let Some(listing_type) = self.listing_type {
+      query = match listing_type {
+        ListingType::Subscribed => query.filter(community_follower::person_id.is_not_null()), // TODO could be this: and(community_follower::person_id.eq(person_id_join)),
+        ListingType::Local => query.filter(community::local.eq(true)),
+        _ => query,
+      };
+    }
+
+    // Don't show blocked communities or nsfw communities if not enabled in profile
+    if self.local_user.is_some() {
+      query = query.filter(community_block::person_id.is_null());
+      query = query.filter(community::nsfw.eq(false).or(local_user::show_nsfw.eq(true)));
+    } else {
+      // No person in request, only show nsfw communities if show_nsfw is passed into request
+      if !self.show_nsfw.unwrap_or(false) {
+        query = query.filter(community::nsfw.eq(false));
+      }
+    }
+
+    let (limit, offset) = limit_and_offset(self.page, self.limit)?;
+    let res = query
+      .limit(limit)
+      .offset(offset)
+      .load::<CommunityExportViewTuple>(conn)
+      .await?;
+
+    Ok(res.into_iter().map(CommunityExportView::from_tuple).collect())
+  }
+}
+
+impl JoinView for CommunityExportView {
+  type JoinTuple = CommunityExportViewTuple;
+  fn from_tuple(a: Self::JoinTuple) -> Self {
+    Self {
+      community: a.0,
+      subscribed: CommunityFollower::to_subscribed_type(&a.1),
+      blocked: a.2.is_some(),
+    }
+  }
+}
```

Comment:
Dont see why this is necessary, I would expect the export to contain only a list of `community.actor_id` strings.